pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  callPackage,
+  buildExamples ? false,
+  buildTests ? false,
+  buildAll ? false,
+}:
+
+# Derivation build example:
+#   $ nix-build -A sobjectizer
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sobjectizer";
+  version = "5.8.4";
+
+  src = fetchFromGitHub {
+    owner = "Stiffstream";
+    repo = "sobjectizer";
+    rev = "48240cff83d758f0e7fc413ec408536147a374ed";
+    hash = "sha256-tIqWgd6TppHfqZk3XHzhG0t+Nn8BQCTP81UD7ls67UE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeDir = "../dev";
+
+  cmakeFlags = lib.optionals (buildExamples) [ "-DBUILD_EXAMPLES=ON" ] ++
+               lib.optionals (buildTests)    [ "-DBUILD_TESTS=ON" ] ++
+               lib.optionals (buildAll)      [ "-DBUILD_ALL=ON" ];
+
+  doCheck = buildTests || buildAll;
+  checkPhase = ''
+    runHook preCheck
+
+    ctest -C Release --output-on-failure
+
+    runHook postCheck
+  '';
+
+  # Test invocation example:
+  #   $ nix-build --attr pkgs.sobjectizer.passthru.tests.ping-pong-*
+  passthru = {
+    tests = {
+      ping-pong-static = callPackage ./tests/ping-pong-static { };
+      ping-pong-shared = callPackage ./tests/ping-pong-shared { };
+    };
```

Comment:
Switching to `with{Static, Shared}` means you'll pass `sobjectizer.override { withStatic = true; };`  and `sobjectizer.override { withShared = true; };`  into these `callPackage` invocations

Reply:
> Switching to `with{Static, Shared}` means you'll pass `sobjectizer.override { withStatic = true; };` and `sobjectizer.override { withShared = true; };` into these `callPackage` invocations

There's really no need for that anymore. Tests that running in `doCheck` is enough.